src/solver/batchnorm/backward_spatial_multiple.cpp

```diff
@@ -38,9 +38,34 @@ namespace solver {
 
 namespace batchnorm {
 
+bool BNBwdIsCaseVariant2(const miopen::batchnorm::ProblemDescription& problem)
+{
+    int n, c, h, w;
+    std::tie(n, c, h, w) = tien<4>(problem.GetXDesc().GetLengths());
+
+    unsigned int in_cstride = h * w;
+    unsigned int in_nhw     = n * in_cstride;
+
+    if(!(in_nhw < (32 * 1024 * 1024) && in_cstride > 1024) &&
+       !(in_nhw < (32 * 1024 * 1024) && in_cstride > 512) && !(in_cstride <= 512))
```

Comment:
It's not that complex as the condition from src/solver/batchnorm/forward_spatial_multiple.cpp, but still can be simplified, since it contains redundant statements.

Reply:
One example of the transformations:
```
X == in_nhw < (32 * 1024 * 1024)
Y == in_cstride > 1024
Z == in_cstride > 512

!(X * Y) & !(X & Z) & !(!Z)
(!X | !Y) & (!X | !Z) & Z
(!X | !Y) & (!X & Z | !Z & Z) // !Z & Z -> false
(!X | !Y) & !X & Z
!X & Z | !Y & !X & Z
!X & Z (1 | !Y)
!X & Z
```
So it basically means  `return (in_nhw >= (32 * 1024 * 1024)) && (in_cstride > 512);`

If there are any doubts about those transformations, here is a proof (tested in excel, lol):
|X|Y|Z|old|new|result|
|-|-|-|-|-|-|
|0|0|0|FALSE|FALSE|TRUE|
|1|0|0|FALSE|FALSE|TRUE|
|0|1|0|FALSE|FALSE|TRUE|
|1|1|0|FALSE|FALSE|TRUE|
|0|0|1|TRUE|TRUE|TRUE|
|1|0|1|FALSE|FALSE|TRUE|
|0|1|1|TRUE|TRUE|TRUE|
|1|1|1|FALSE|FALSE|TRUE|